pkg/lib/html2po.py

```diff
@@ -0,0 +1,147 @@
+#
+# Note that some of the use of the translated may not support all the strings
+# depending on the code actually using these strings to translate the HTML.
+
+import argparse
+import os
+import sys
+from typing import Dict
+
+import bs4
+import polib
+
+
+def scan_tags(filename, entries, soup):
+    """
+    Look for tags marked as translatable and extract them
+    """
+
+    def is_translatable(tag):
+        return tag.has_attr("translate") or tag.has_attr("translatable")
+
+    tags = soup.find_all(is_translatable)
+
+    # Extract translate strings
+    for tag in tags:
+        tasks = (
+            tag.attrs.get("translate") or tag.attrs.get("translatable") or "yes"
+        ).split(" ")
+
+        # Calculate the line location
+        line = tag.sourceline or 0
+
+        msgctxt = tag.attrs.get("translate-context") or tag.attrs.get("context")
+        msgid_plural = tag.attrs.get("translate-plural")
+        location = (filename, str(line))
+
+        # For each thing listed
+        for task in tasks:
+            msgid = None
+            # The element text itself
+            if task == "yes" or task == "translate":
+                msgid = tag.get_text(strip=True)
+            # An attribute
+            elif task:
+                msgid = tag.attrs[task]
+
+            if msgid:
+                push(
+                    entries,
+                    msgid=msgid,
+                    msgid_plural=msgid_plural,
+                    msgctxt=msgctxt,
+                    location=location,
+                )
+
+
+def push(entries, msgid, msgid_plural, msgctxt, location):
+    """
+    Push an entry onto the list
+    """
+    key = msgid + "\0" + (msgid_plural or "") + "\0" + (msgctxt or "")
+    if key in entries:
+        entries[key].occurrences.append(location)
+    else:
+        entries[key] = polib.POEntry(
+            msgid=msgid,
+            msgid_plural=msgid_plural,
+            msgctxt=msgctxt,
+            occurrences=[location],
+        )
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        prog="html2po",
+        description="Extracts translatable strings from HTML files",
+    )
+    parser.add_argument("-d", "--directory", help="Base directory for input files")
+    parser.add_argument("-o", "--output", help="Output file", required=True)
+    parser.add_argument(
+        "files", nargs="+", help="One or more input files", metavar="FILE"
+    )
+
+    args = parser.parse_args()
+
+    # The unique messages extracted from the files
+    entries: Dict[str, polib.POEntry] = {}
+
+    # Now process each file in turn
+    for filename in args.files:
+        # Qualify the filename if necessary
+        full = filename
+        if args.directory:
+            full = os.path.join(args.directory, filename)
+
+        with open(full) as f:
+            soup = bs4.BeautifulSoup(f, "html.parser")
+        scan_tags(filename, entries, soup)
+
+    po = polib.POFile()
+    po.metadata = {
```

Comment:
Note the static code check failure here:
```
not ok 7 /static-code/test-pyvulture
# pkg/lib/html2po.py:130: unused attribute 'metadata' (60% confidence)
```
This is obviously bogus, this is a "magic" object. This needs to be quiesced somehow.

This may disappear with the change away from polib, let's discuss that in PR #18737